crates/next-api/src/nft_json.rs

```diff
@@ -0,0 +1,191 @@
+    }
+
+    #[turbo_tasks::function]
+    async fn get_output_specifier(
+        self: Vc<Self>,
+        path: Vc<FileSystemPath>,
+    ) -> Result<Vc<OutputSpecifier>> {
+        let this = self.await?;
+        let path_fs = path.fs().resolve().await?;
+        let path_ref = path.await?;
+        let nft_folder = self.ident().path().parent().await?;
+
+        if path_fs == Vc::upcast(this.output_fs.resolve().await?) {
+            // e.g. a referenced chunk
+            return Ok(Vc::cell(Some(
+                nft_folder.get_relative_path_to(&path_ref).unwrap(),
+            )));
+        } else if path_fs == Vc::upcast(this.project_fs.resolve().await?) {
+            return Ok(Vc::cell(Some(
+                self.ident_in_project_fs()
+                    .await?
+                    .get_relative_path_to(&path_ref)
+                    .unwrap(),
+            )));
+        } else if path_fs == Vc::upcast(this.client_fs.resolve().await?) {
+            return Ok(Vc::cell(Some(
+                self.ident_in_client_fs()
+                    .await?
+                    .get_relative_path_to(&path_ref)
+                    .unwrap()
+                    .replace("/_next/", "/.next/")
+                    .into(),
+            )));
+        }
+
+        if let Some(path_fs) = Vc::try_resolve_downcast_type::<VirtualFileSystem>(path_fs).await? {
+            if path_fs.await?.name == "externals" || path_fs.await?.name == "traced" {
+                return Ok(Vc::cell(Some(
+                    self.ident_in_project_fs()
+                        .await?
+                        .get_relative_path_to(
+                            &*this.project_fs.root().join(path_ref.path.clone()).await?,
+                        )
+                        .unwrap(),
+                )));
+            }
+        }
+
+        println!("Unknown filesystem for {}", path.to_string().await?);
+        Ok(Vc::cell(None))
+    }
+}
+
+#[turbo_tasks::value_impl]
+impl OutputAsset for NftJsonAsset {
+    #[turbo_tasks::function]
+    async fn ident(&self) -> Result<Vc<AssetIdent>> {
+        let path = self.chunk.ident().path().await?;
+        Ok(AssetIdent::from_path(
+            path.fs
+                .root()
+                .join(format!("{}.nft.json", path.path).into()),
+        ))
+    }
+}
+
+#[turbo_tasks::value_impl]
+impl Asset for NftJsonAsset {
+    #[turbo_tasks::function]
+    async fn content(self: Vc<Self>) -> Result<Vc<AssetContent>> {
+        let this = &*self.await?;
+        let mut result = Vec::new();
+
+        let chunk = this.chunk.to_resolved().await?;
+        let entries = this
+            .additional_assets
+            .iter()
+            .copied()
+            .chain(std::iter::once(chunk))
+            .collect();
+        for referenced_chunk in all_assets_from_entries(Vc::cell(entries)).await? {
+            if referenced_chunk.ident().path().await?.extension_ref() == Some("map") {
```

Comment:
Should source maps be included in lambdas? We should discuss that with the build&deploy team.

Reply:
Created a ticket and linked it here, will update